File: backend/src/main/java/com/example/aiprompt/domain/model/Conversation.java
package com.example.aiprompt.domain.model;

import java.time.Instant;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.UUID;

public class Conversation {

    private final UUID id;
    private final Instant createdAt;
    private final List<Message> messages;

    private Conversation(UUID id, Instant createdAt, List<Message> messages) {
        this.id = id;
        this.createdAt = createdAt;
        this.messages = new ArrayList<>(messages);
    }

    public static Conversation create() {
        return new Conversation(UUID.randomUUID(), Instant.now(), Collections.emptyList());
    }

    public UUID getId() {
        return id;
    }

    public Instant getCreatedAt() {
        return createdAt;
    }

    public List<Message> getMessages() {
        return Collections.unmodifiableList(messages);
    }

    public ConversationStatus getStatus() {
        return messages.isEmpty() ? ConversationStatus.EMPTY : ConversationStatus.ACTIVE;
    }

    public void addMessagePair(Message userMessage, Message assistantMessage) {
        if (userMessage == null || userMessage.role() != MessageRole.USER) {
            throw new IllegalArgumentException("First message must be a USER message");
        }
        if (assistantMessage == null || assistantMessage.role() != MessageRole.ASSISTANT) {
            throw new IllegalArgumentException("Second message must be an ASSISTANT message");
        }
        messages.add(userMessage);
        messages.add(assistantMessage);
    }
}

File: frontend/lib/types/conversation.ts
export enum MessageRole {
  USER = "USER",
  ASSISTANT = "ASSISTANT",
}

export enum ConversationStatus {
  EMPTY = "EMPTY",
  ACTIVE = "ACTIVE",
}

export interface MessageResponse {
  id: string
  role: MessageRole
  content: string
  timestamp: string
}

export interface HateoasLink {
  href: string
  method?: string
}

export interface HateoasLinks {
  self?: HateoasLink
  "send-prompt"?: HateoasLink
  clear?: HateoasLink
  new?: HateoasLink
  conversation?: HateoasLink
  [key: string]: HateoasLink | undefined
}

export interface ConversationResponse {
  id: string
  status: ConversationStatus
  messages: MessageResponse[]
  createdAt: string
  _links: HateoasLinks
}

export interface PromptResultResponse {
  conversationId: string
  userMessage: MessageResponse
  assistantMessage: MessageResponse
  _links: HateoasLinks
}
